shopfloor/services/service.py

```diff
@@ -194,19 +197,40 @@ def _to_json(self, records):
             res.append(self._convert_one_record(record))
         return res
 
-    def _get_input_validator(self, method_name):
-        # override the method to get the validator in a component
-        # instead of a method, to keep things apart
-        validator_component = self.component(usage="%s.validator" % self._usage)
-        return validator_component._get_validator(method_name)
-
-    def _get_output_validator(self, method_name):
-        # override the method to get the validator in a component
-        # instead of a method, to keep things apart
+    def _get_validator_schema(self, method_name, usage_suffix):
         validator_component = self.component(
-            usage="%s.validator.response" % self._usage
+            usage="{}.{}".format(self._usage, usage_suffix)
         )
-        return validator_component._get_validator(method_name)
+        return getattr(validator_component, method_name)
+
+    def __getattr__(self, item):
+        # We have delegated the validator / return validators to dedicated
```

Comment:
can you add a `FIXME:` here?
Also, maybe is worth to involve @lmignon in the discussion on how to support this in core base_rest?

Reply:
working on this here https://github.com/OCA/rest-framework/pull/99

Reply:
and here https://github.com/OCA/wms/commit/00dd96bba4d69d5e07ec3797e2792e5880e0c4df

Reply:
fixme added